### ISSUE TITLE
Unable to create 50000 branches

### ISSUE THREAD

**Issue by nkonev** [dolthub/dolt#5135]
I am creating a prototype with company registry. Users can create draft of company, edith the draft, but not to touch main version, and when it is ok they will approve it to be applied.
So I use branch for drafts, merge for merge. (You can use my prototype for guides or how-tos for JVM example (Kotlin + Liquibase + Spring Data JDBC), but it's the different story :) )

I created so-called load test, which tries to create 50_000 branches. It's unable to accomplish this because of error below.

The version is 0.52.0.

**Reproducer** (need JDK 17):
```
git clone git@github.com:nkonev/companies-many-branches.git
cd companies-many-branches
git checkout repro
docker-compose up -d
./mvnw -DenableLoadTests=true -Dtest="name.nkonev.examples.companies.VolumeTest#fifty_thousands_branches" test
# wait about of hour
```

After the some time at JVM side I see
```
2023-01-12 18:16:39.551  INFO 66153 --- [           main] n.nkonev.examples.companies.VolumeTest   : 32763 of 50000 - Created branch with name a50e4744-851a-41bf-b88a-5941da9b551b
2023-01-12 18:16:39.634  INFO 66153 --- [           main] n.nkonev.examples.companies.VolumeTest   : 32764 of 50000 - Created branch with name 4590ffc9-0535-4c2d-b8b3-339d0dddba17
2023-01-12 18:16:39.714  INFO 66153 --- [           main] n.nkonev.examples.companies.VolumeTest   : 32765 of 50000 - Created branch with name 1f485a7e-1a9f-4bd3-9dcd-5be5a2565b64
2023-01-12 18:16:39.801  INFO 66153 --- [           main] n.nkonev.examples.companies.VolumeTest   : 32766 of 50000 - Created branch with name dd769a4a-6ea7-4302-a34d-0d6d55e27477
2023-01-12 18:16:39.880  INFO 66153 --- [           main] n.nkonev.examples.companies.VolumeTest   : 32767 of 50000 - Created branch with name 037e1da9-23a2-418b-83c8-381a69c7d640
2023-01-12 18:16:39.988  WARN 66153 --- [           main] com.zaxxer.hikari.pool.ProxyConnection   : HikariPool-1 - Connection com.mysql.cj.jdbc.ConnectionImpl@1ca73158 marked as broken because of SQLSTATE(08S01), ErrorCode(0)

com.mysql.cj.jdbc.exceptions.CommunicationsException: Communications link failure

The last packet successfully received from the server was 102 milliseconds ago. The last packet sent successfully to the server was 102 milliseconds ago.
        at com.mysql.cj.jdbc.exceptions.SQLError.createCommunicationsException(SQLError.java:174) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:64) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:916) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1061) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1009) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1320) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:994) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61) ~[HikariCP-4.0.3.jar:na]
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java) ~[HikariCP-4.0.3.jar:na]
        at org.springframework.jdbc.core.JdbcTemplate.lambda$update$2(JdbcTemplate.java:965) ~[spring-jdbc-5.3.23.jar:5.3.23]
        at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:651) ~[spring-jdbc-5.3.23.jar:5.3.23]
        at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:960) ~[spring-jdbc-5.3.23.jar:5.3.23]
        at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:1015) ~[spring-jdbc-5.3.23.jar:5.3.23]
        at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:1025) ~[spring-jdbc-5.3.23.jar:5.3.23]
        at name.nkonev.examples.companies.StorageService.createAndCheckoutBranch(StorageService.kt:37) ~[classes/:na]
        at name.nkonev.examples.companies.VolumeTest.fifty_thousands_branches(VolumeTest.kt:34) ~[test-classes/:na]
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
        at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[na:na]
        at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725) ~[junit-platform-commons-1.8.2.jar:1.8.2]
        at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66) ~[junit-jupiter-engine-5.8.2.jar:5.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511) ~[na:na]
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511) ~[na:na]
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54) ~[junit-platform-engine-1.8.2.jar:1.8.2]
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220) ~[junit-platform-launcher-1.3.1.jar:1.3.1]
        at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188) ~[junit-platform-launcher-1.3.1.jar:1.3.1]
        at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202) ~[junit-platform-launcher-1.3.1.jar:1.3.1]
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181) ~[junit-platform-launcher-1.3.1.jar:1.3.1]
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128) ~[junit-platform-launcher-1.3.1.jar:1.3.1]
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:150) ~[surefire-junit-platform-2.22.2.jar:2.22.2]
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:124) ~[surefire-junit-platform-2.22.2.jar:2.22.2]
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384) ~[surefire-booter-2.22.2.jar:2.22.2]
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345) ~[surefire-booter-2.22.2.jar:2.22.2]
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126) ~[surefire-booter-2.22.2.jar:2.22.2]
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418) ~[surefire-booter-2.22.2.jar:2.22.2]
Caused by: com.mysql.cj.exceptions.CJCommunicationsException: Communications link failure

The last packet successfully received from the server was 102 milliseconds ago. The last packet sent successfully to the server was 102 milliseconds ago.
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:na]
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77) ~[na:na]
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:na]
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[na:na]
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480) ~[na:na]
        at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:61) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:105) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.exceptions.ExceptionFactory.createException(ExceptionFactory.java:151) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.exceptions.ExceptionFactory.createCommunicationsException(ExceptionFactory.java:167) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.protocol.a.NativeProtocol.readMessage(NativeProtocol.java:581) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.protocol.a.NativeProtocol.checkErrorMessage(NativeProtocol.java:761) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.protocol.a.NativeProtocol.sendCommand(NativeProtocol.java:700) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.protocol.a.NativeProtocol.sendQueryPacket(NativeProtocol.java:1051) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.NativeSession.execSQL(NativeSession.java:665) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:893) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        ... 78 common frames omitted
Caused by: java.io.EOFException: Can not read response from server. Expected to read 4 bytes, read 0 bytes before connection was unexpectedly lost.
        at com.mysql.cj.protocol.FullReadInputStream.readFully(FullReadInputStream.java:67) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.protocol.a.SimplePacketReader.readHeaderLocal(SimplePacketReader.java:81) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.protocol.a.SimplePacketReader.readHeader(SimplePacketReader.java:63) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.protocol.a.SimplePacketReader.readHeader(SimplePacketReader.java:45) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.protocol.a.TimeTrackingPacketReader.readHeader(TimeTrackingPacketReader.java:52) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.protocol.a.TimeTrackingPacketReader.readHeader(TimeTrackingPacketReader.java:41) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.protocol.a.MultiPacketReader.readHeader(MultiPacketReader.java:54) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.protocol.a.MultiPacketReader.readHeader(MultiPacketReader.java:44) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        at com.mysql.cj.protocol.a.NativeProtocol.readMessage(NativeProtocol.java:575) ~[mysql-connector-j-8.0.31.jar:8.0.31]
        ... 83 common frames omitted

[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1,581.19 s <<< FAILURE! - in name.nkonev.examples.companies.VolumeTest
[ERROR] fifty_thousands_branches  Time elapsed: 1,577.27 s  <<< ERROR!
org.springframework.dao.RecoverableDataAccessException: 
PreparedStatementCallback; SQL [CALL DOLT_CHECKOUT ('-b', ?)]; Communications link failure

The last packet successfully received from the server was 102 milliseconds ago. The last packet sent successfully to the server was 102 milliseconds ago.; nested exception is com.mysql.cj.jdbc.exceptions.CommunicationsException: Communications link failure

The last packet successfully received from the server was 102 milliseconds ago. The last packet sent successfully to the server was 102 milliseconds ago.
        at name.nkonev.examples.companies.VolumeTest.fifty_thousands_branches(VolumeTest.kt:34)
Caused by: com.mysql.cj.jdbc.exceptions.CommunicationsException: 
Communications link failure

The last packet successfully received from the server was 102 milliseconds ago. The last packet sent successfully to the server was 102 milliseconds ago.
        at name.nkonev.examples.companies.VolumeTest.fifty_thousands_branches(VolumeTest.kt:34)
Caused by: com.mysql.cj.exceptions.CJCommunicationsException: 
Communications link failure

The last packet successfully received from the server was 102 milliseconds ago. The last packet sent successfully to the server was 102 milliseconds ago.
        at name.nkonev.examples.companies.VolumeTest.fifty_thousands_branches(VolumeTest.kt:34)
Caused by: java.io.EOFException: Can not read response from server. Expected to read 4 bytes, read 0 bytes before connection was unexpectedly lost.
        at name.nkonev.examples.companies.VolumeTest.fifty_thousands_branches(VolumeTest.kt:34)

2023-01-12 18:16:40.372  INFO 66153 --- [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown initiated...
2023-01-12 18:16:40.374  INFO 66153 --- [ionShutdownHook] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Shutdown completed.
```

In docker logs of dolt
```
[nkonev@MiWiFi-R4A-srv ~]$ docker logs -f --tail 400 companies-many-branches_doltdb_1
2023-01-12 13:50:02+00:00 [Note] [Entrypoint]: Entrypoint script for Dolt Server 0.52.0 starting.
2023-01-12 13:50:02+00:00 [Note] [Entrypoint]: Running init scripts
2023-01-12 13:50:02+00:00 [Note] [Entrypoint]: /usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/init.sql


Starting server with Config HP="0.0.0.0:3306"|T="28800000"|R="false"|L="info"
2023-01-12T13:50:21Z INFO [conn 1] NewConnection {DisableClientMultiStatements=false}
2023-01-12T13:50:21Z INFO [conn 2] NewConnection {DisableClientMultiStatements=false}
2023-01-12T13:50:21Z INFO [conn 3] NewConnection {DisableClientMultiStatements=false}
2023-01-12T13:50:21Z INFO [conn 4] NewConnection {DisableClientMultiStatements=false}
2023-01-12T13:50:21Z INFO [conn 5] NewConnection {DisableClientMultiStatements=false}
2023-01-12T13:50:21Z INFO [conn 6] NewConnection {DisableClientMultiStatements=false}
2023-01-12T13:50:21Z INFO [conn 7] NewConnection {DisableClientMultiStatements=false}
2023-01-12T13:50:21Z INFO [conn 8] NewConnection {DisableClientMultiStatements=false}
2023-01-12T13:50:21Z INFO [conn 9] NewConnection {DisableClientMultiStatements=false}
2023-01-12T13:50:21Z INFO [conn 10] NewConnection {DisableClientMultiStatements=false}
2023-01-12T13:50:21Z WARN [conn 1] error running query {connectTime=2023-01-12T13:50:21Z, connectionDb=companies, error=table not found: DATABASECHANGELOGLOCK, query=SELECT COUNT(*) FROM companies.DATABASECHANGELOGLOCK}
2023-01-12T14:16:39Z INFO [no conn] kill query: pid 32913 {}
2023-01-12T14:16:39Z INFO [conn 1] ConnectionClosed {}
2023-01-12T14:16:39Z ERROR [no conn] mysql_server caught panic:
message is too large to be encoded
/src/gen/fb/serial/fileidentifiers.go:72 (0xa74a8f)
/src/libraries/doltcore/branch_control/branch_control.go:161 (0x1552b2f)
/src/libraries/doltcore/branch_control/branch_control.go:283 (0x1553c04)
/src/libraries/doltcore/env/actions/branch.go:203 (0x156e313)
/src/libraries/doltcore/sqle/dprocedures/dolt_checkout.go:234 (0x17777ba)
/src/libraries/doltcore/sqle/dprocedures/dolt_checkout.go:78 (0x1776729)
/src/libraries/doltcore/sqle/dprocedures/dolt_checkout.go:37 (0x17762a4)
/usr/local/go/src/reflect/value.go:584 (0x4c17c4)
/usr/local/go/src/reflect/value.go:368 (0x4c0cbb)
/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.14.1-0.20230103223835-f83cb10bd54f/sql/plan/external_procedure.go:131 (0x13ceed1)
/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.14.1-0.20230103223835-f83cb10bd54f/sql/plan/procedure.go:172 (0x13af6c6)
/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.14.1-0.20230103223835-f83cb10bd54f/sql/plan/call.go:173 (0x13af68c)
/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.14.1-0.20230103223835-f83cb10bd54f/sql/plan/transaction_committing_iter.go:63 (0x14312e5)
/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.14.1-0.20230103223835-f83cb10bd54f/sql/plan/process.go:66 (0x14035d7)
/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.14.1-0.20230103223835-f83cb10bd54f/engine.go:280 (0x166a9cd)
/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.14.1-0.20230103223835-f83cb10bd54f/server/handler.go:352 (0x176c749)
/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.14.1-0.20230103223835-f83cb10bd54f/server/handler.go:616 (0x176f3c7)
/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.14.1-0.20230103223835-f83cb10bd54f/server/handler.go:176 (0x176a8c5)
/go/pkg/mod/github.com/dolthub/vitess@v0.0.0-20221221215514-0a479c28185a/go/mysql/conn.go:1288 (0xc240af)
/go/pkg/mod/github.com/dolthub/vitess@v0.0.0-20221221215514-0a479c28185a/go/mysql/conn.go:930 (0xc20e78)
/go/pkg/mod/github.com/dolthub/vitess@v0.0.0-20221221215514-0a479c28185a/go/mysql/server.go:481 (0xc3a3b2)
/usr/local/go/src/runtime/asm_amd64.s:1594 (0x46bf80)
 {}
2023-01-12T14:16:39Z INFO [conn 11] NewConnection {DisableClientMultiStatements=false}
2023-01-12T14:16:40Z INFO [conn 2] ConnectionClosed {}
2023-01-12T14:16:40Z INFO [conn 4] ConnectionClosed {}
2023-01-12T14:16:40Z INFO [conn 3] ConnectionClosed {}
2023-01-12T14:16:40Z INFO [conn 5] ConnectionClosed {}
2023-01-12T14:16:40Z INFO [conn 6] ConnectionClosed {}
2023-01-12T14:16:40Z INFO [conn 9] ConnectionClosed {}
2023-01-12T14:16:40Z INFO [conn 7] ConnectionClosed {}
2023-01-12T14:16:40Z INFO [conn 10] ConnectionClosed {}
2023-01-12T14:16:40Z INFO [conn 8] ConnectionClosed {}
2023-01-12T14:16:40Z INFO [conn 11] ConnectionClosed {}
```

UPD:
So, having the last succcessful line
```
32767 of 50000
```
I can conclude that right now the number of branches limited by 32767, like in Java's short type: The short data type is a 16-bit signed two's complement integer. It has a minimum value of -32,768 and a maximum value of 32,767 (inclusive)

**Comment by fulghum** [comment]
Hey @nkonev, thanks for trying out Dolt! This looks like an interesting load test. Thank you for sharing the results and log output with us! Is 50k branches about what you expect for your application? Just looking to understand what your app needs so we can help make Dolt work for you. 

The log output you included is helpful. From digging through the Dolt stack trace, it looks like the error is not directly with creating the branch itself, but with setting up some admin metadata for the branch. We persist that admin metadata to disk as flatbuffer encoded data and that's when we hit that panic about the flatbuffer message size being too large. 

@Hydrocharged – do you have more context on this code? My first guess is that the branch control access table has just grown too large after all these branches are added?

**Comment by nkonev** [comment]
Thanks @fulghum 
So the estimated number of branches in bad case can be 70_000, in the worst case it can be 140_000. Personally I wanna believe that this case will never happen, but I need to ensure that our app will run correctly even in bad and in the worst cases.

**Comment by Hydrocharged** [comment]
It's definitely coming from branch control. My only real guess is that flatbuffers is using an `int16` to encode an array's length, which doesn't _seem_ like they would choose such a small number to encode an array's items, but I'm not 100% sure. Our only real limitation in branch control is that branch names need to fit within 2¹⁶-1 characters, and that would result in truncation rather than a panic. I'll look into flatbuffers and see if I can't pinpoint something a bit more concrete.

**Comment by Hydrocharged** [comment]
@nkonev I've been able to replicate this locally, and in summary we're running into a file size limit that we've hardcoded into one of the file formats that we store information in. There are two sides to this that I'm now looking into:
- Are we able to raise the limit without detriment?
- Why are we hitting the limit in the first place?
An answer to these will inform me of the best way to fix this, but do know that a fix is coming soon as I continue the investigation!

**Comment by Hydrocharged** [comment]
@nkonev A fix is in!
https://github.com/dolthub/dolt/pull/5167
We’ll do a release later today so that you can work with the fix.

**Comment by nkonev** [comment]
fixed in 0.52.8, thank you!

**Comment by Hydrocharged** [comment]
Fantastic! You’re welcome 😄
